docs/adr/0035-refactor-payments-frontend.md

```diff
@@ -0,0 +1,101 @@
+# Refactor Subscription Platform frontend
```

Comment:
Nit: might add "with NestJS" in the title and MD file name to match our others (and so the primary decision is obvious when looking at the ADR file tree)